src/Compilers/CSharp/Test/Symbol/Symbols/LocalFunctionTests.cs

```diff
@@ -189,5 +189,181 @@ public void M()
+                    Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureVariable, "c").WithArguments("c").WithLocation(10, 30),
+                    // (17,17): error CS8820: A static anonymous function cannot contain a reference to 'c'.
+                    //         var x = c.MyExtension;
+                    Diagnostic(ErrorCode.ERR_StaticAnonymousFunctionCannotCaptureVariable, "c").WithArguments("c").WithLocation(17, 17),
+                    // (18,34): error CS8820: A static anonymous function cannot contain a reference to 'c'.
+                    //         var y = new Func<string>(c.MyExtension);
+                    Diagnostic(ErrorCode.ERR_StaticAnonymousFunctionCannotCaptureVariable, "c").WithArguments("c").WithLocation(18, 34));
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup2()
+        {
+            CreateCompilation("""
+                using System;
+
+                var c = new C();
+                LocalFunc();
+                NonStatic();
+
+                static void LocalFunc()
+                {
+                    var x = Extensions.MyExtension;
+                    var y = new Func<C, string>(Extensions.MyExtension);
+                }
+
+                void NonStatic()
+                {
+                    Action f = static () =>
+                    {
+                        var x = Extensions.MyExtension;
+                        var y = new Func<C, string>(Extensions.MyExtension);
+                    };
+                }
+
+                public class C
+                {
+                }
+
+                public static class Extensions
+                {
+                    public static string MyExtension(this C c)
+                        => string.Empty;
+                }
+                """).VerifyDiagnostics().VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup3()
+        {
+            CreateCompilation("""
+                using System;
+                public class Base { }
+
+                public class C : Base
+                {
+                    public void M()
+                    {
+                        LocalFunc();
+                        NonStatic();
+
+                        static void LocalFunc()
+                        {
+                            var x1 = this.MyExtension;
+                            var x2 = new Func<string>(this.MyExtension);
+                            var y1 = base.MyExtension;
+                            var y2 = new Func<string>(base.MyExtension);
+                        }
+                
+                        void NonStatic()
+                        {
+                            Action f = static () =>
+                            {
+                                var x1 = this.MyExtension;
+                                var x2 = new Func<string>(this.MyExtension);
```

Comment:
Are we reporting errors for these cases? (Consider renaming the locals in this lambda to `x3`, `x4`, `y3`, `y4` so it's easier to match up diagnostics.)